prow/plugins/lgtm/lgtm.go

```diff
@@ -263,35 +266,68 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	// Only add the label if it doesn't have it, and vice versa.
 	labels, err := gc.GetIssueLabels(org, repoName, number)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repoName, number)
+		log.WithError(err).Errorf("Failed to get issue labels.")
 	}
 	hasLGTM := github.HasLabel(LGTMLabel, labels)
 
 	// remove the label if necessary, we're done after this
 	if hasLGTM && !wantLGTM {
 		log.Info("Removing LGTM label.")
-		return gc.RemoveLabel(org, repoName, number, LGTMLabel)
-	}
-
-	// add the label if necessary, we're done after this
-	if !hasLGTM && wantLGTM {
+		if err := gc.RemoveLabel(org, repoName, number, LGTMLabel); err != nil {
+			return err
+		}
+		botname, err := gc.BotName()
+		if err != nil {
+			return err
+		}
+		comments, err := gc.ListIssueComments(org, repoName, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get issue comments.")
+		}
+		for _, comment := range comments {
+			m := addLGTMLabelNotificationRe.FindStringSubmatch(comment.Body)
+			if comment.User.Login == botname && m != nil {
+				log.Info("Removing tree-hash comment.")
+				if err := gc.DeleteComment(org, repoName, comment.ID); err != nil {
```

Comment:
The comment deletion can be simplified and made more efficient if you use the `CommentPruner` client in the plugin client. You can follow the pattern used in the [`require-matching-label` plugin](https://github.com/kubernetes/test-infra/blob/b2e00a416ee0f71aabee5802074637061de4f175/prow/plugins/require-matching-label/require-matching-label.go). The `shouldPrune` function may assume that all bot comments it considers are bot comments.

Reply:
I have done so, but I need your (@cjwagner ) help to fix the unit tests... I have found nowhere an example on how to actually fake PruneComments deleting comments.